src/main/java/htsjdk/samtools/cram/structure/Container.java

```diff
@@ -59,10 +63,57 @@
      */
     public long offset;
 
+    /**
+     * Set this container's sequence ID and derive its {@link SliceType} from that value
+     * @param sequenceId the sequence ID
+     */
+    public Container(final int sequenceId) {
+        this.sequenceId = sequenceId;
+        this.containerType = SliceType.bySequenceId(sequenceId);
+    }
+
+    public static Container newMultiRefContainer() {
+        return new Container(Slice.REF_ID_MULTIPLE);
+    }
+
+    public static Container newUnmappedContainer() {
+        return new Container(Slice.REF_ID_UNMAPPED);
+    }
+
+    public SliceType getContainerType() {
+        return containerType;
+    }
+
+    /**
+     * Get this container's sequence ID or sentinel value, suitable for serialization:
+     * 0 or greater for a single reference
+     * Slice.REF_ID_MULTIPLE (-2) for a multiple reference slice
+     * Slice.REF_ID_UNMAPPED (-1) for an unmapped slice
+     * @return the sequence ID
+     */
+    public int getSequenceIdOrSentinel() {
+        return sequenceId;
+    }
+
+    /**
+     * Get this single-reference container's valid sequence ID
+     * @throw CRAMException if this container is not single-ref
+     * @return the sequence ID
+     */
+    public int getValidSequenceId() {
```

Comment:
How about `getReferenceSequenceID` ?

Reply:
Open to changing the name but I would like some signifier that this also validates

Reply:
Ok - I was thinking that "valid" should be implied, and that if a qualifier is needed somewhere, it should be on the method that *isn't* valid (i.e., what is currently called `getSequenceIdOrSentinel`, which might return a validID or it might return a sentinel).

Thinking out loud here though, I would argue we should seek to not have `getSequenceIdOrSentinel` at all now that we have `SliceType` and `ReferenceContext`. Part of the goal is to eliminate the need for code (at least outside of `Slice` and `Container`) from needing to overload integer sequence ID variables with sentinel values. Ideally the sentinels could be kept private.

Maybe `SliceType` should be called `ReferenceContextType`; and `ReferenceContext` should have a constructor that takes a `ReferenceContextType` and one that takes an integer ID. Then evrything would be type-safe, and his method could be called `getReferenceSequence`, which would only be valid if the `ReferenceContextType` for a `ReferenceContext` is SINGLE_REFERENCE.